core/src/test/java/org/ocpsoft/PrettyTimeI18n_SO_Test.java

```diff
@@ -0,0 +1,508 @@
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.ocpsoft.prettytime;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.temporal.ChronoUnit;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.List;
+import java.util.Locale;
+
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Test;
+import org.ocpsoft.prettytime.format.SimpleTimeFormat;
+import org.ocpsoft.prettytime.impl.ResourcesTimeFormat;
+import org.ocpsoft.prettytime.units.Minute;
+
+public class PrettyTimeTest
+{
+   private Locale locale;
+   private LocalDateTime now;
+
+   @Before
+   public void setUp() throws Exception
+   {
+      locale = Locale.getDefault();
+      Locale.setDefault(new Locale("so","SO"));
+      now = LocalDateTime.now();
+   }
+
+   @Test
+   public void testCeilingInterval() throws Exception
+   {
+      PrettyTime prettyTime = new PrettyTime(LocalDate.of(2009, 6, 17));
+      Assert.assertEquals("1 bil kahor", prettyTime.format(LocalDate.of(2009, 5, 20)));
+   }
+
+   @Test
+   public void testNullDate() throws Exception
+   {
+      PrettyTime t = new PrettyTime();
+      Date date = null;
+      Assert.assertEquals("waxyar kadib", t.format(date));
+   }
+
+   @Test
+   public void testRightNow() throws Exception
+   {
+      PrettyTime t = new PrettyTime();
+      Assert.assertEquals("waxyar kadib", t.format(new Date()));
+   }
+
+   @Test
+   public void testRightNowWithReference() throws Exception
+   {
+      Date now = new Date();
+      PrettyTime t = new PrettyTime();
+      Assert.assertEquals("waxyar kadib", t.format(now));
+   }
+
+   @Test
+   public void testCalculatePreciceDuration() throws Exception
+   {
+      PrettyTime t = new PrettyTime();
+      List<Duration> preciseDuration = t.calculatePreciseDuration(
+               new Date(System.currentTimeMillis() - (2 * 60 * 60 * 1000) - (2 * 60 * 1000)));
+      Assert.assertEquals("2 saac 2 daqiiqo kahor", t.format(preciseDuration));
+      Assert.assertEquals("2 saac 2 daqiiqo", t.formatDuration(preciseDuration));
+      Assert.assertEquals("waxyar kahor",
+               t.format(t.calculatePreciseDuration(new Date(new Date().getTime() - 10))));
+
+      preciseDuration = t.calculatePreciseDuration(now.minusHours(2).minusMinutes(2));
+      Assert.assertEquals("2 saac 2 daqiiqo kahor", t.format(preciseDuration));
+      Assert.assertEquals("2 saac 2 daqiiqo", t.formatDuration(preciseDuration));
+      Assert.assertEquals("waxyar kahor", t.format(t.calculatePreciseDuration(LocalDateTime.now().plusSeconds(1))));
+   }
+
+   @Test
+   @SuppressWarnings("Digniin")
```

Comment:
This probably needs to be 'deprecation' like it was previously. I don't think this will compile in the EN_US locale. (Where it's built & published to Maven)